CMake/OtherTests.cmake

```
@@ -132,3 +133,69 @@ if(NOT CMAKE_CROSSCOMPILING)
     }" HAVE_POLL_FINE)
   endif()
 endif()
+
+# Detect HAVE_GETADDRINFO_THREADSAFE
+
+if(WIN32)
+  set(HAVE_GETADDRINFO_THREADSAFE ${HAVE_GETADDRINFO})
+elseif(CMAKE_SYSTEM_NAME STREQUAL "AIX" OR
```

note autotools only checks the OSes if HAVE_GETADDRINFO
https://github.com/curl/curl/blob/6fa1d817e5b1a00d7d0c8168091877476b499317/m4/curl-functions.m4#L2011-L2014

so it might need a check like 
```suggestion
elseif(NOT HAVE_GETADDRINFO)
  set(HAVE_GETADDRINFO_THREADSAFE FALSE)
elseif(CMAKE_SYSTEM_NAME STREQUAL "AIX" OR
```

---

Thank you!